pkg/engine/armvariables_test.go

```diff
@@ -124,7 +124,7 @@ func TestK8sVars(t *testing.T) {
 		"nsgName":                        "[concat(variables('masterVMNamePrefix'), 'nsg')]",
 		"orchestratorNameVersionTag":     "Kubernetes:" + testK8sVersion,
 		"primaryAvailabilitySetName":     "",
-		"primaryScaleSetName":            "[concat(parameters('orchestratorName'), '-agentpool1-',parameters('nameSuffix'), '-vmss')]",
+		"primaryScaleSetName":            cs.Properties.GetPrimaryScaleSetName(),
```

Comment:
This is the actual fix.

Reply:
In other words, we're validating that the value of the ARM variable `primaryScaleSetName` derives from the canonical `primaryScaleSetName()` method, not from an ARM expression string.